sql/solutions/lesson4.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: page
+title: Introduction to SQL Solutions
+---
+# Lesson 1 exercises Solutions
+
+## L4.1 Update the age of the person named Luis Marco Polo.
+
+Solution 1:
+First find person's id:
+```SQL
+SELECT *
+  FROM people
+ WHERE name = 'Luis Marco Polo';
+```
+Update age using id number:
+```SQL
+UPDATE people
+   SET age = age + 1
+ WHERE id = 3;
+```
+
+Solution 2:
+Just use their name as identifier to update age. (When could this cause problems?)
+```SQL
+UPDATE people
+   SET age = age + 1
+ WHERE name = 'Luis Marco Polo';
+```
+## L4.2 Increase by 1 the age of all the people who were born in Italy.
+
+```SQL
+UPDATE people
+   SET age = age + 1
+ WHERE country_of_birth = 'IT';
+```
+
+## L4.3 Set all names to lowercase.
+
+```SQL
+UPDATE people
+   SET name = LOWER(name);
+```
+
+## L4.4 Insert a new student.
+
+Solution 1:
+What problems could arise with this solution?
+```SQL
+INSERT INTO people (id, name)
+VALUES (42, 'Valerie Knight');
```

Comment:
Although this is correct because all the other columns allow NULL, I think to help student in the learning process is better to put more columns. Do you agree?

Reply:
It's likely interesting for them to run the query both ways -- to see how you can fill in all the values and to see how a db will populate missing values as 'NULL' when you haven't done a complete job.

Reply:
Wrote the solution both ways :)